client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx

```diff
@@ -263,6 +265,19 @@ const DomainsStep: Step = function DomainsStep( { navigation, flow } ) {
 		submitWithDomain( suggestion );
 	};
 
+	const onUseYourDomainClick = ( domain?: string ) => {
+		if ( isHundredYearPlanFlow( flow ) || isHundredYearDomainFlow( flow ) ) {
```

Comment:
Would that do the same thing?

```
if ( domain && isHundredYearDomainFlow( flow ) ) {
    const leaveFlowFunction = exitFlow ?? window.location.assign;
    leaveFlowFunction( `/setup/hundred-year-domain-transfer?new=${ domain }` );
}

setShowUseYourDomain( true );
```

I think it is easier to read and doesn't affect the hundred year plan.